### PR TITLE
Remove OIDC from code

### PR DESCRIPTION
**Pull request by arielshaqed**
Was deprecated... and it's gone!

### CODE REVIEW

File: pkg/api/auth_middleware.go [hunk 1]
```diff
@@ -2,10 +2,8 @@ package api
 
 import (
```
Comment (Isan-Rivkin) [inline pkg/api/auth_middleware.go:3]: not compiling unused resources

**Comment by arielshaqed** [comment]
## Note to releaser:

Breaking change, requires minor version bump!

File: api/swagger.yml [hunk 2]
```diff
@@ -29,10 +28,6 @@ components:
       type: apiKey
       in: cookie
       name: internal_auth_session
-    oidc_auth:
-      type: apiKey
-      in: cookie
-      name: oidc_auth_session
```
Comment (Isan-Rivkin) [inline api/swagger.yml:35]: we need this, that's the cookie that is verified by lakeFS.

File: pkg/api/auth_middleware.go [hunk 3]
```diff
@@ -117,13 +112,7 @@ func checkSecurityRequirements(r *http.Request,
 					continue
 				}
 				user, err = userByToken(ctx, logger, authService, token)
-			case "oidc_auth":
```
Comment (Isan-Rivkin) [inline pkg/api/auth_middleware.go:120]: we need this  + `userFromOIDC`

**Review by Isan-Rivkin — CHANGES_REQUESTED** [review]

**Comment by ortz** [comment]
@arielshaqed I'll go over it later today, we need someone from VE to be on the reviewers list too (we had a decision back then that shared-ownership repositories changes should be reviewed by at least 1 participant from each team)

**Comment by arielshaqed** [comment]
Restored the missing code, thanks!  A private image works on a manually-hacked cloud dev-env.  I tested explicit logout as well as deleted OIDC auth cookie.

PTAL.

File: api/swagger.yml (hunk 2 above)
Comment (arielshaqed) [reply api/swagger.yml:35]: Restored!

**Review by arielshaqed — COMMENTED** [review]
Thanks!

PTAL...

**Review by Isan-Rivkin — APPROVED** [review]
@arielshaqed  Looking good! 
Please don't forget that this release dependent on a new helm/lakefs release so that the configurations will not break :)

**Review by itaiad200 — APPROVED** [review]

**Comment by arielshaqed** [comment]
Rebased on trunk; will auto-merge once tests pass.

**Pull request merged**